spec/services/tag_importer/links_publisher_spec.rb

```diff
@@ -0,0 +1,153 @@
+require 'rails_helper'
+
+RSpec.describe TagImporter::LinksPublisher do
```

Comment:
Yup, especially the setup for each scenario...

I will try to make it better without writing an integration test first if that's ok. If that doesn't work I might split the work into more chunks.